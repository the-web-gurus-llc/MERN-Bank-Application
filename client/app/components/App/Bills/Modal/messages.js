import { defineMessages } from 'react-intl';

export const scope = 'app.components.Modal';

export default defineMessages({
  create: {
    id: `${scope}.create`,
    defaultMessage: 'Create',
  },
  cancel: {
    id: `${scope}.cancel`,
    defaultMessage: 'Cancel',
  },
  title: {
    id: `${scope}.title`,
    defaultMessage: 'Create a new bill',
  },
  descriptionTop: {
    id: `${scope}.description.top`,
    defaultMessage:
      'You are currently trying to create a new bill. Creating a new bill is free.',
  },
  currencyRequired: {
    id: `${scope}.currency.required`,
    defaultMessage: 'Currency is required.',
  },
  placeholder: {
    id: `${scope}.select.placeholder`,
    defaultMessage: 'Select currency',
  },
  descriptionBottom: {
    id: `${scope}.description.bottom`,
    defaultMessage: 'Remember that the number of bills you have is limited.',
  },
  billHasBeenCreated: {
    id: `${scope}.bill.has.been.created`,
    defaultMessage: 'Bill has been created',
  },
  billHasBeenCreatedDescription: {
    id: `${scope}.bill.has.been.created.description`,
    defaultMessage:
      'New bill has been assigned to your account and appeared in your bills widget.',
  },
  billHasNotBeenCreated: {
    id: `${scope}.bill.has.not.been.created`,
    defaultMessage: 'Bill has not been created',
  },
  billHasNotBeenCreatedDescription: {
    id: `${scope}.bill.has.not.been.created.description`,
    defaultMessage:
      'You cannot create a new account because the maximum number of accounts has been exceeded.',
  },
});
