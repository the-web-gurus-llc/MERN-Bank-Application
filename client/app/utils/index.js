export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
