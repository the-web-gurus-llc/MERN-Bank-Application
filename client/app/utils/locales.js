exports.DEFAULT_LOCALE = 'en';
exports.DEFAULT_DATE_FORMAT = 'dd.MM.yyyy, hh:mm aa';

// prettier-ignore
exports.appLocales = [
  'en',
  'de',
  'pl',
];
