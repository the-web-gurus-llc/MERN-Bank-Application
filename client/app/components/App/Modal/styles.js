import styled from 'styled-components';
import { Modal } from 'antd';
import { media } from 'utils';

export const StyledModal = styled(Modal)`
  p {
    &:last-child {
      margin-bottom: 0;
    }
  }

  .ant-modal-footer {
    display: flex;
    justify-content: flex-end;
    align-items: center;

    button {
      width: 50%;

      div {
        white-space: pre-wrap;
      }
    }
  }

  ${media.tablet`
    .ant-modal-footer {

      button {
        width: auto;
      }
    } 
  `}
`;
