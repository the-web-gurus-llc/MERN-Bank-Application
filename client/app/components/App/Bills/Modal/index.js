import React from 'react';
import { Button } from 'antd';
import { StyledForm } from 'components/Form/styles';
import { createStructuredSelector } from 'reselect';
import {
  makeSelectIsOpenedModal,
  makeSelectCurrency,
} from 'containers/DashboardPage/selectors';
import { useSelector, useDispatch } from 'react-redux';
import CurrencyToggle from 'components/CurrencyToggle';
import { createNewBillAction } from 'containers/DashboardPage/actions';
import LoadingIndicator from 'components/LoadingIndicator';
import { makeSelectIsLoading } from 'providers/LoadingProvider/selectors';
import { getRequestName } from 'helpers';
import { CREATE_NEW_BILL_REQUEST } from 'containers/DashboardPage/constants';
import { injectIntl, intlShape, FormattedMessage } from 'react-intl';
import { toggleModalAction } from 'containers/App/actions';
import { StyledModal } from 'components/App/Modal/styles';
import messages from './messages';

const stateSelector = createStructuredSelector({
  currency: makeSelectCurrency(),
  isOpenedModal: makeSelectIsOpenedModal(),
  isLoading: makeSelectIsLoading(getRequestName(CREATE_NEW_BILL_REQUEST)),
});

function Modal({ intl }) {
  const [form] = StyledForm.useForm();
  const dispatch = useDispatch();
  const { isOpenedModal, isLoading } = useSelector(stateSelector);
  const snippets = {
    success: {
      title: intl.formatMessage(messages.billHasBeenCreated),
      description: intl.formatMessage(messages.billHasBeenCreatedDescription),
    },
    error: {
      title: intl.formatMessage(messages.billHasNotBeenCreated),
      description: intl.formatMessage(
        messages.billHasNotBeenCreatedDescription,
      ),
    },
  };

  const createNewBill = () => dispatch(createNewBillAction(snippets));
  const toggleModal = () => {
    dispatch(toggleModalAction());
    form.resetFields();
  };

  const onValidateFields = async () => {
    try {
      await form.validateFields();
      createNewBill();
      toggleModal();
      form.resetFields();
    } catch (error) {
      Error(error);
    }
  };

  return (
    <StyledModal
      title={intl.formatMessage(messages.title)}
      visible={isOpenedModal}
      onOk={onValidateFields}
      onCancel={toggleModal}
      centered="true"
      footer={[
        <Button key="back" onClick={toggleModal}>
          <FormattedMessage {...messages.cancel} />
        </Button>,
        <Button
          key="submit"
          disabled={isLoading}
          type="primary"
          onClick={onValidateFields}
        >
          {isLoading ? (
            <LoadingIndicator />
          ) : (
            <FormattedMessage {...messages.create} />
          )}
        </Button>,
      ]}
    >
      <>
        <p>
          <FormattedMessage {...messages.descriptionTop} />
        </p>

        <StyledForm centered="true" form={form} name="create-new-bill">
          <CurrencyToggle tailed="true" />
        </StyledForm>

        <p>
          <FormattedMessage {...messages.descriptionBottom} />
        </p>
      </>
    </StyledModal>
  );
}

Modal.propTypes = {
  intl: intlShape.isRequired,
};

export default injectIntl(Modal);
