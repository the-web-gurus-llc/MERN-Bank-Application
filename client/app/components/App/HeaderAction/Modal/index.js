import React from 'react';
import { FormattedMessage } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import { useDispatch, useSelector } from 'react-redux';
import { makeSelectIsOpenedModal } from 'containers/App/selectors';
import { makeSelectIsLoading } from 'providers/LoadingProvider/selectors';
import { getRequestName } from 'helpers';
import { LOGOUT_REQUEST } from 'containers/App/constants';
import { logoutAction, toggleConfirmModalAction } from 'containers/App/actions';
import { StyledModal } from 'components/App/Modal/styles';
import { Button } from 'antd';
import LoadingIndicator from 'components/LoadingIndicator';
import messages from '../messages';

const stateSelector = createStructuredSelector({
  isOpenedModal: makeSelectIsOpenedModal(),
  isLoading: makeSelectIsLoading(getRequestName(LOGOUT_REQUEST)),
});

export default function Modal() {
  const { isOpenedModal, isLoading } = useSelector(stateSelector);
  const dispatch = useDispatch();

  const onLogout = () => dispatch(logoutAction());
  const onToggleConfirmModal = () => dispatch(toggleConfirmModalAction());

  return (
    <FormattedMessage {...messages.logout}>
      {(title) => (
        <StyledModal
          title={title}
          visible={isOpenedModal}
          onOk={onLogout}
          onCancel={onToggleConfirmModal}
          footer={[
            <Button key="back" onClick={onToggleConfirmModal}>
              <FormattedMessage {...messages.popConfirmCancel} />
            </Button>,
            <Button
              key="submit"
              disabled={isLoading}
              type="primary"
              onClick={onLogout}
            >
              {isLoading ? (
                <LoadingIndicator />
              ) : (
                <FormattedMessage {...messages.popConfirmOk} />
              )}
            </Button>,
          ]}
        >
          <FormattedMessage {...messages.popConfirmTitle} />
        </StyledModal>
      )}
    </FormattedMessage>
  );
}
