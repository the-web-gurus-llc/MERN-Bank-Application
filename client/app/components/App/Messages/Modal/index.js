/* eslint-disable react/no-danger */
import React from 'react';
import { StyledModal } from 'components/App/Modal/styles';
import { createStructuredSelector } from 'reselect';
import { useSelector, useDispatch } from 'react-redux';
import { makeSelectLocale } from 'providers/LanguageProvider/selectors';
import {
  makeSelectIsOpenedMessage,
  makeSelectOpenedMessage,
  makeSelectMessages,
} from 'containers/App/selectors';
import { StyledButton } from 'components/App/Button/styles';
import { closeMessageModalAction } from 'containers/App/actions';

const stateSelector = createStructuredSelector({
  isOpenedMessage: makeSelectIsOpenedMessage(),
  openedMessage: makeSelectOpenedMessage(),
  messages: makeSelectMessages(),
  locale: makeSelectLocale(),
});

export default function Modal() {
  const { isOpenedMessage, openedMessage, messages, locale } = useSelector(
    stateSelector,
  );
  const dispatch = useDispatch();

  const onCloseMessageModal = () => dispatch(closeMessageModalAction());

  const subject = messages?.data
    ?.find((message) => message.uuid === openedMessage)
    ?.templates.find((template) => template.language.code === locale)?.subject;
  const content = messages?.data
    ?.find((message) => message.uuid === openedMessage)
    ?.templates.find((template) => template.language.code === locale)?.content;
  const actions = messages?.data
    ?.find((message) => message.uuid === openedMessage)
    ?.templates.find((template) => template.language.code === locale)?.actions;

  return (
    <StyledModal
      title={subject}
      visible={isOpenedMessage}
      onOk={onCloseMessageModal}
      onCancel={onCloseMessageModal}
      centered="true"
      footer={actions?.map((action, index) => (
        <StyledButton
          key={index}
          hovered="true"
          type="link"
          onClick={onCloseMessageModal}
        >
          <div dangerouslySetInnerHTML={{ __html: action }}></div>
        </StyledButton>
      ))}
    >
      <div dangerouslySetInnerHTML={{ __html: content }} />
    </StyledModal>
  );
}
