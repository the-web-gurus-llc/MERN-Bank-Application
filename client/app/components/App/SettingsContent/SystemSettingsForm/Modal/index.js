import React from 'react';
import LoadingIndicator from 'components/LoadingIndicator';
import { StyledModal } from 'components/App/Modal/styles';
import { Button } from 'antd';
import { FormattedMessage } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { setUserDataAction } from 'containers/SettingsPage/actions';
import { SET_USER_DATA_REQUEST } from 'containers/SettingsPage/constants';
import { getRequestName } from 'helpers';
import { makeSelectIsLoading } from 'providers/LoadingProvider/selectors';
import { toggleModalAction } from 'containers/App/actions';
import { makeSelectCurrencies } from 'containers/App/selectors';
import {
  makeSelectIsOpenedModal,
  makeSelectNewData,
} from 'containers/SettingsPage/selectors';
import messages from '../messages';

const stateSelector = createStructuredSelector({
  currencies: makeSelectCurrencies(),
  newData: makeSelectNewData(),
  isOpenedModal: makeSelectIsOpenedModal(),
  isLoading: makeSelectIsLoading([getRequestName(SET_USER_DATA_REQUEST)]),
});

export default function Modal({ snippets, form }) {
  const dispatch = useDispatch();
  const { isLoading, isOpenedModal, currencies, newData } = useSelector(
    stateSelector,
  );

  const onToggleModal = () => {
    form.resetFields();
    dispatch(toggleModalAction());
  };
  const onSetUserData = () => dispatch(setUserDataAction(snippets));

  return (
    <FormattedMessage {...messages.title}>
      {(title) => (
        <StyledModal
          centered="true"
          title={title}
          visible={isOpenedModal}
          onOk={onSetUserData}
          onCancel={onToggleModal}
          footer={[
            <Button key="back" onClick={onToggleModal}>
              <FormattedMessage {...messages.cancel} />
            </Button>,
            <Button
              key="submit"
              disabled={isLoading}
              type="primary"
              onClick={onSetUserData}
            >
              {isLoading ? (
                <LoadingIndicator />
              ) : (
                <FormattedMessage {...messages.confirm} />
              )}
            </Button>,
          ]}
        >
          <p>
            <FormattedMessage
              {...messages.tryingChangeDefaultCurrencyPartOne}
            />{' '}
            <strong>
              {
                currencies?.find(
                  (currency) => currency?.uuid === newData?.currency,
                )?.name
              }
            </strong>
            <FormattedMessage
              {...messages.tryingChangeDefaultCurrencyPartTwo}
            />
          </p>
          <p>
            <FormattedMessage
              {...messages.tryingChangeDefaultCurrencyConfirm}
            />
          </p>
        </StyledModal>
      )}
    </FormattedMessage>
  );
}

Modal.propTypes = {
  snippets: PropTypes.shape({
    success: PropTypes.shape({
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    }),
  }).isRequired,
  form: PropTypes.object.isRequired,
};
