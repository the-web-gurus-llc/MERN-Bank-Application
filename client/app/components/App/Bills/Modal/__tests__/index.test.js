import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'utils/configureStore';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'utils/locales';
import Modal from '../index';
import 'utils/__tests__/__mocks__/matchMedia';

describe('<Modal />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a Modal', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Modal />
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
