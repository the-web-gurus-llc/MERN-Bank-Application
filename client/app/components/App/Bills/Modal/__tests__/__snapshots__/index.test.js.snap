// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render a Modal 1`] = `null`;
